### PR TITLE
feat/update-tokens-balance

### PR DESCRIPTION
**Pull request by leozera14**
Feat: Implement method to update ERC tokens balance at getLatestUpdate

### CODE REVIEW

File: source/scripts/Background/controllers/account/syscoin.ts [hunk 1]
```diff
@@ -99,61 +104,99 @@ const SysAccountController = (): ISysAccountController => {
...
+                label: accounts[index].label,
+                transactions: [...filteredTxs],
+                assets: {
+                  ethereum: accounts[index].assets.ethereum,
+                  syscoin: account.assets,
+                },
+              };
+            else {
+              //UPDATE ETH ERC TOKEN BALANCES
+              const getUpdatedErcTokens = await getBalanceUpdatedToErcTokens(
```
Comment (claudiovb) [inline source/scripts/Background/controllers/account/syscoin.ts:179]: Also is possible to check here the dispatch of setAccounts, which is only triggered on this function is also losing information regarding the Ethereum state is possible to check in the below print:
image
It needs to be reviewed as well.

File: source/utils/tokens.ts [hunk 2]
```diff
@@ -7,3 +14,62 @@ export const getSymbolByChain = async (chain: string) => {
...
+          ? balance
+          : lodash.floor(parseFloat(balance as string), 4);
+
+        return { ...vaultAssets, balance: formattedBalance };
+      })
+    );
+
+    return updatedTokens;
+  } catch (error) {
+    return [];
```
Comment (claudiovb) [inline source/utils/tokens.ts:73]: This catch probably should return the old balance of the assets, as RPCS tends to go down sometimes and also we need to know the error. Something similar to this:
`... catch (error){
 console.error('Pali utils: Couldn't update assets due to the following issue ', error);
....
return { ...vaultAssets }
}`

File: source/utils/tokens.ts [hunk 3]
```diff
@@ -7,3 +14,62 @@ export const getSymbolByChain = async (chain: string) => {
...
+
+    const callBalance = await provider.getBalance(accountAddress);
+
+    const balance = ethers.utils.formatEther(callBalance);
+
+    const formattedBalance = lodash.floor(parseFloat(balance), 4);
+
+    return formattedBalance;
+  } catch (error) {
+    return 0;
```
Comment (claudiovb) [inline source/utils/tokens.ts:33]: The same comment for getBalanceUpdatedToErcTokens function applies here

File: source/state/vault/index.ts [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import { createSlice, PayloadAction } from '@reduxjs/toolkit';
 import { ethers } from 'ethers';
 import loadsh from 'lodash';
+import lodash from 'lodash';
```
Comment (claudiovb) [inline source/state/vault/index.ts:4]: There seem to be two repeated imports from lodash here

**Review by claudiovb — CHANGES_REQUESTED** [review]
The feature is working properly when sending tokens from the wallet UI, it updated the two balances involved in the transaction.
But the comments outlined within this review must be addressed before we are able to merge this PR

File: source/state/vault/index.ts (hunk 4 above)
Comment (leozera14) [reply source/state/vault/index.ts]: fixed

File: source/utils/tokens.ts (hunk 3 above)
Comment (leozera14) [reply source/utils/tokens.ts]: fixed

File: source/utils/tokens.ts (hunk 2 above)
Comment (leozera14) [reply source/utils/tokens.ts]: fixed

**Review by claudiovb — APPROVED** [review]

**Pull request merged**